pkg/asset/k8s.go

```diff
@@ -116,16 +118,38 @@ const validBootstrapTokenChars = "0123456789abcdefghijklmnopqrstuvwxyz"
 // newBootstrapToken constructs a bootstrap token in conformance with the following format:
 // https://kubernetes.io/docs/admin/bootstrap-tokens/#token-format
 func newBootstrapToken() (id string, secret string, err error) {
-	// Read 6 random bytes for the id and 16 random bytes for the token (see spec for details).
-	token := make([]byte, 6+16)
-	if _, err := rand.Read(token); err != nil {
+	idBytes := make([]byte, 6)
+	secretBytes := make([]byte, 16)
+	tokenLength := len(idBytes) + len(secretBytes)
+	charsetLength := len(validBootstrapTokenChars)
+	size := int(math.Ceil(float64(tokenLength) * math.Log2(float64(charsetLength)) / math.Log2(256)))
+
+	random := make([]byte, size)
+	if _, err := rand.Read(random); err != nil {
 		return "", "", err
 	}
 
-	for i, b := range token {
-		token[i] = validBootstrapTokenChars[int(b)%len(validBootstrapTokenChars)]
+	encodeBytes(validBootstrapTokenChars, random, idBytes, secretBytes)
```

Comment:
Don't have this be a function call? It just renames the variable and makes it harder to follow.

Reply:
> Don't have this be a function call?

It's a function call so you can test base 256 -> base `len(charset)` conversion without pulling in `randRead`.  If you don't want the test, I can inline the conversion code here.